### PR TITLE
Remove alpha csi CRD

### PR DESCRIPTION
**Pull request by msau42**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
Alpha storage CRDs are no longer supported.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
CSI alpha CRDs have been removed
```

### CODE REVIEW

File: cluster/gce/gci/configure-helper.sh [hunk 1]
```diff
@@ -2638,9 +2638,6 @@ EOF
   if [[ "${ENABLE_DEFAULT_STORAGE_CLASS:-}" == "true" ]]; then
     setup-addon-manifests "addons" "storage-class/gce"
   fi
-  if [[ "${FEATURE_GATES:-}" =~ "AllAlpha=true"  || "${FEATURE_GATES:-}" =~ "CSIDriverRegistry=true" || "${FEATURE_GATES:-}" =~ "CSINodeInfo=true" ]]; then
-    setup-addon-manifests "addons" "storage-crds"
-  fi
```
Comment (bertinatto) [inline cluster/gce/gci/configure-helper.sh]: You might want to remove the install commands from `hack/local-up-cluster.sh` as well

Comment (msau42) [reply cluster/gce/gci/configure-helper.sh]: good catch, removed

**Comment by msau42** [comment]
Updated to also remove the files in the addons directory

File: staging/publishing/rules.yaml [hunk 2]
```diff
@@ -669,64 +669,6 @@ rules:
         branch: release-1.14
       - repository: client-go
         branch: release-11.0
-- destination: csi-api
```
Comment (msau42) [inline staging/publishing/rules.yaml]: I hope this is fine

**Comment by saad-ali** [comment]
Some build file clean up still remaining (causing test failure).

Otherwise LGTM

/lgtm
/approve

File: staging/publishing/rules.yaml (hunk 2 above)
Comment (liggitt) [reply staging/publishing/rules.yaml]: needs rebasing, this file now only contains the rules for the master branch, so removing for 1.15 would be fine

**Comment by liggitt** [comment]
lgtm, needs rebase

**Comment by msau42** [comment]
Rebased

**Comment by liggitt** [comment]
/lgtm

@sttts, fyi

**Pull request merged**